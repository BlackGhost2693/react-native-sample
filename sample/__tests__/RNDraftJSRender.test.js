import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';

import RNDraftJSRender from '../../index';
import data from '../src/resourceMock.json';

it('renders correctly with contentState', () => {
  const tree = renderer.create(
    <RNDraftJSRender contentState={data} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with contentState and customStyle', () => {
  const customStyles = {
    unstyled: {
      fontSize: 18,
    },
    italic: {
      fontSize: 20,
    },
  };
  const tree = renderer.create(
    <RNDraftJSRender contentState={data} customStyles={customStyles} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
