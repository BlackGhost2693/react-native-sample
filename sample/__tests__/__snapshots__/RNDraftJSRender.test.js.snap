// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with contentState 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 22,
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    Maecenas nec odio
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Etiam ultricies nisi vel 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      augue
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      . Sed magna purus, fermentum eu, tincidunt eu, varius ut, felis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; In ac dui quis mi consectetuer lacinia. Cras dapibus. Sed 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontStyle": "italic",
          },
          undefined,
        ]
      }
    >
      mollis
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      , eros et ultrices tempus, mauris ipsum aliquam libero, non 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      adipiscing
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       dolor urna a orci.
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    Aenean leo ligula porttitor
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 19,
          "fontWeight": "bold",
          "marginBottom": 19,
          "marginTop": 19,
        },
        undefined,
      ]
    }
  >
    Morbi vestibulum volutpat
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 21,
          "marginTop": 21,
        },
        undefined,
      ]
    }
  >
    Class aptent taciti sociosqu
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 13,
          "fontWeight": "bold",
          "marginBottom": 22,
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    Phasellus leo dolor
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 11,
          "fontWeight": "bold",
          "marginBottom": 25,
          "marginTop": 25,
        },
        undefined,
      ]
    }
  >
    Donec vitae orci sed
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      1
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Proin magna. Donec posuere vulputate arcu
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      2
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Etiam feugiat lorem non metus
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      3
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
       Curabitur at lacus ac velit ornare lobortis
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 5,
            "height": 5,
            "marginRight": 8,
            "width": 5,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Phasellus a est.
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 5,
            "height": 5,
            "marginRight": 8,
            "width": 5,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Fusce ac felis sit amet ligula pharetra condimentum
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 5,
            "height": 5,
            "marginRight": 8,
            "width": 5,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Suspendisse 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "textDecorationLine": "underline",
            },
            undefined,
          ]
        }
      >
        faucibus
      </Text>
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#eee",
          "borderLeftWidth": 4,
          "borderStyle": "solid",
          "marginBottom": 22,
          "marginTop": 22,
          "paddingLeft": 12,
        },
        undefined,
      ]
    }
  >
    <View
      style={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontStyle": "italic",
            "fontWeight": "normal",
            "marginLeft": 16,
          },
          undefined,
        ]
      }
    >
      Suspendisse potenti. Fusce a quam. Etiam ut purus mattis mauris sodales aliquam. Morbi vestibulum volutpat enim. Phasellus nec sem in justo pellentesque facilisis.
    </Text>
    <View
      style={undefined}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "#cecece",
          "fontFamily": "Courier New",
          "padding": 16,
        },
        undefined,
      ]
    }
  >
    'code-block': {
      backgroundColor: '#cecece',
      fontFamily: 'Courier New',
      padding: 16,
    },
  </Text>
</View>
`;

exports[`renders correctly with contentState and customStyle 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 22,
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    Maecenas nec odio
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "normal",
        },
        Object {
          "fontSize": 18,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Etiam ultricies nisi vel 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      augue
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      . Sed magna purus, fermentum eu, tincidunt eu, varius ut, felis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; In ac dui quis mi consectetuer lacinia. Cras dapibus. Sed 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontStyle": "italic",
          },
          Object {
            "fontSize": 20,
          },
        ]
      }
    >
      mollis
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      , eros et ultrices tempus, mauris ipsum aliquam libero, non 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      adipiscing
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       dolor urna a orci.
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    Aenean leo ligula porttitor
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 19,
          "fontWeight": "bold",
          "marginBottom": 19,
          "marginTop": 19,
        },
        undefined,
      ]
    }
  >
    Morbi vestibulum volutpat
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 21,
          "marginTop": 21,
        },
        undefined,
      ]
    }
  >
    Class aptent taciti sociosqu
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 13,
          "fontWeight": "bold",
          "marginBottom": 22,
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    Phasellus leo dolor
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 11,
          "fontWeight": "bold",
          "marginBottom": 25,
          "marginTop": 25,
        },
        undefined,
      ]
    }
  >
    Donec vitae orci sed
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      1
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Proin magna. Donec posuere vulputate arcu
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      2
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Etiam feugiat lorem non metus
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 12,
            "fontWeight": "bold",
            "marginRight": 8,
          },
          undefined,
        ]
      }
    >
      3
      .
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
       Curabitur at lacus ac velit ornare lobortis
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 5,
            "height": 5,
            "marginRight": 8,
            "width": 5,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Phasellus a est.
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 5,
            "height": 5,
            "marginRight": 8,
            "width": 5,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      Fusce ac felis sit amet ligula pharetra condimentum
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 5,
            "height": 5,
            "marginRight": 8,
            "width": 5,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Suspendisse 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Array [
            Object {
              "textDecorationLine": "underline",
            },
            undefined,
          ]
        }
      >
        faucibus
      </Text>
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderLeftColor": "#eee",
          "borderLeftWidth": 4,
          "borderStyle": "solid",
          "marginBottom": 22,
          "marginTop": 22,
          "paddingLeft": 12,
        },
        undefined,
      ]
    }
  >
    <View
      style={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontStyle": "italic",
            "fontWeight": "normal",
            "marginLeft": 16,
          },
          undefined,
        ]
      }
    >
      Suspendisse potenti. Fusce a quam. Etiam ut purus mattis mauris sodales aliquam. Morbi vestibulum volutpat enim. Phasellus nec sem in justo pellentesque facilisis.
    </Text>
    <View
      style={undefined}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "#cecece",
          "fontFamily": "Courier New",
          "padding": 16,
        },
        undefined,
      ]
    }
  >
    'code-block': {
      backgroundColor: '#cecece',
      fontFamily: 'Courier New',
      padding: 16,
    },
  </Text>
</View>
`;
